docs/src/main/sphinx/connector/bigquery.rst

```diff
@@ -141,30 +141,130 @@ Property                                              Description
-============== =============================== =============================================================================================================
+.. _bigquery-type-mapping:
+
+Type mapping
+------------
+
+Because Trino and BigQuery each support types that the other does not, this
+connector modifies some types when reading or writing data. Data types may not
+map the same way in both directions between Trino and the data source. Refer to
+the following sections for type mapping in each direction.
+
+BigQuery type to Trino type mapping
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The connector maps BigQuery types to the corresponding Trino types according
+to the following table:
+
+.. list-table:: BigQuery type to Trino type mapping
+  :widths: 30, 25, 50
+  :header-rows: 1
+
+  * - BigQuery database type
+    - Trino type
+    - Notes
+  * - ``ARRAY``
+    - ``ARRAY``
+    -
+  * - ``BOOLEAN``
+    - ``BOOLEAN``
+    -
+  * - ``BYTES``
+    - ``VARBINARY``
+    -
+  * - ``DATE``
+    - ``DATE``
+    -
+  * - ``DATETIME``
+    - ``TIMESTAMP(6)``
+    -
+  * - ``FLOAT``
+    - ``DOUBLE``
+    -
+  * - ``GEOGRAPHY``
+    - ``VARCHAR``
+    - In `Well-known text (WKT) <https://en.wikipedia.org/wiki/Well-known_text_representation_of_geometry>`_ format
+  * - ``INTEGER``
+    - ``BIGINT``
+    -
+  * - ``NUMERIC``
+    - ``DECIMAL(P,S)``
+    - The default precision and scale of ``NUMERIC`` is ``(38, 9)``.
+  * - ``BIGNUMERIC``
+    - ``DECIMAL(P,S)``
+    - Precision > 38 is not supported. The default precision and scale of
+      ``BIGNUMERIC`` is ``(77, 38)``.
+  * - ``RECORD``
+    - ``ROW``
+    -
+  * - ``STRING``
+    - ``VARCHAR``
+    -
+  * - ``TIME``
+    - ``TIME(6)``
+    -
+  * - ``TIMESTAMP``
+    - ``TIMESTAMP(6) WITH TIME ZONE``
+    - Time zone is UTC
+
+No other types are supported.
+
+Trino type to BigQuery type mapping
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The connector maps Trino types to the corresponding BigQuery types according
+to the following table:
+
+.. list-table:: Trino type to BigQuery type mapping
+  :widths: 30, 25, 50
+  :header-rows: 1
+
+  * - Trino type
+    - BigQuery type
+    - Notes
+  * - ``ARRAY``
+    - ``ARRAY``
+    -
+  * - ``BOOLEAN``
+    - ``BOOLEAN``
+    -
+  * - ``VARBINARY``
+    - ``BYTES``
+    -
+  * - ``DATE``
+    - ``DATE``
+    -
+  * - ``DATETIME``
+    - ``TIMESTAMP(6)``
+    -
+  * - ``DOUBLE``
+    - ``FLOAT``
+    -
+  * - ``VARCHAR``
+    - ``GEOGRAPHY``
+    - In `Well-known text (WKT) <https://en.wikipedia.org/wiki/Well-known_text_representation_of_geometry>`_ format
+  * - ``BIGINT``
+    - ``INTEGER``
+    -
+  * - ``DECIMAL(P,S)``
+    - ``BIGNUMERIC``
+    - Precision > 38 is not supported. The default precision and scale of
+      ``BIGNUMERIC`` is ``(77, 38)``.
+  * - ``RECORD``
+    - ``ROW``
+    -
+  * - ``VARCHAR``
+    - ``STRING``
+    -
+  * - ``TIME(6)``
+    - ``TIME``
+    -
+  * - ``TIMESTAMP(6) WITH TIME ZONE``
+    - ``TIMESTAMP``
```

Comment:
Remove. Inserting `TIME` and `TIMESTAMP` rows is unsupported.

Reply:
Will do. Thank you kindly for your thorough review @ebyhr.